### PR TITLE
Refactor frontend

### PR DESCRIPTION
**Pull request by Julian702**
## Description

This pr refactors the frontend as described in #143

### CODE REVIEW

File: frontend/components/navigation/Tabs/DeckSettingsTab.tsx [hunk 1]
```diff
@@ -36,8 +38,8 @@ export const DeckSettingsTab = ({
 	const router = useRouter();
 	const { mutate } = useSWRConfig();
 
-	const [deckState, setDeck] = useState(deck);
-	const [isConfirmDeletion, setConfirmDelete] = useState(false);
+	const [deckState, setDeck] = useState<Deck>(deck);
```
Comment (4KevR) [inline frontend/components/navigation/Tabs/DeckSettingsTab.tsx:41]: Inconsistent name

**Review by 4KevR — APPROVED** [review]

**Review by memeToasty — APPROVED** [review]

**Review by TomRomeo — APPROVED** [review]
💖

File: frontend/components/form/InputField.tsx [hunk 2]
```diff
@@ -66,7 +66,7 @@ interface InputFieldProps {
 	 */
 	minLength?: number;
 	/**
-	 * pattern the value has to match to be valid
+	 * Pattern the value has to match to be valid
```
Comment (JJthere) [inline frontend/components/form/InputField.tsx:69]: What is this sentence supposed to mean?

**Review by darmiel — APPROVED** [review]

File: frontend/components/group/Member.tsx [hunk 3]
```diff
@@ -26,7 +26,11 @@ interface MemberProps {
 /**
  * UI component for dislpaying a user
```
Comment (JJthere) [inline frontend/components/group/Member.tsx:27]:
```suggestion
 * UI component for displaying a user
```

File: frontend/components/navigation/Tabs/TabBar.tsx [hunk 4]
```diff
@@ -4,19 +4,19 @@ import { Text } from "../../Text";
...
 	 */
 	id: string;
 	/**
-	 * tabs to display
+	 * Tabs to display
 	 */
 	tabs: { [tab: string]: ReactNode };
 	/**
-	 * currentTab
+	 * CurrentTab
```
Comment (JJthere) [inline frontend/components/navigation/Tabs/TabBar.tsx:15]:
```suggestion
	 * Current tab
```

File: frontend/components/navigation/Tabs/TabBar.tsx [hunk 5]
```diff
@@ -4,19 +4,19 @@ import { Text } from "../../Text";
...
 	 */
 	currentTab: string;
 	/**
-	 * setTab
+	 * SetTab
```
Comment (JJthere) [inline frontend/components/navigation/Tabs/TabBar.tsx:19]:
```suggestion
	 * Set tab
```

File: frontend/components/statistics/Statistic.tsx [hunk 6]
```diff
@@ -9,23 +9,23 @@ import {
...
+	 * Unique identifier
 	 */
 	id: string;
 	/**
 	 * Header
 	 */
 	header: string;
 	/**
-	 * current Value
+	 * Current Value
```
Comment (JJthere) [inline frontend/components/statistics/Statistic.tsx:20]:
```suggestion
	 * Current value
```

File: frontend/components/statistics/Statistic.tsx [hunk 7]
```diff
@@ -9,23 +9,23 @@ import {
...
 	 */
 	value: string;
 	/**
-	 * seperator displayed between value and reference
+	 * Seperator displayed between value and reference
 	 */
 	seperator?: string;
 	/**
-	 * reference Value
+	 * Reference Value
```
Comment (JJthere) [inline frontend/components/statistics/Statistic.tsx:28]:
```suggestion
	 * Reference value
```

**Review by JJthere — CHANGES_REQUESTED** [review]
Well done!
Just found a few nits.

File: frontend/components/statistics/Statistic.tsx [hunk 8]
```diff
@@ -9,23 +9,23 @@ import {
...
 	 */
 	header: string;
 	/**
-	 * current Value
+	 * Current Value
 	 */
 	value: string;
 	/**
-	 * seperator displayed between value and reference
+	 * Seperator displayed between value and reference
```
Comment (leonie2003) [inline frontend/components/statistics/Statistic.tsx:24]: Seperator should be Separator.

**Review by leonie2003 — CHANGES_REQUESTED** [review]
Found just one typo.

File: frontend/components/form/InputField.tsx (hunk 2 above)
Comment (Julian702) [reply frontend/components/form/InputField.tsx:69]: The sentence means that the values has to match the pattern to be a valid input

File: frontend/components/statistics/Statistic.tsx (hunk 8 above)
Comment (Julian702) [reply frontend/components/statistics/Statistic.tsx]: Good catch! Fixed in 54d8b4f7736a62f6952ea8489e07e45cadad1644

File: frontend/components/form/InputField.tsx (hunk 2 above)
Comment (JJthere) [reply frontend/components/form/InputField.tsx:69]: Then why don't you write that?

File: frontend/components/navigation/Tabs/TabBar.tsx (hunk 4 above)
Comment (Julian702) [reply frontend/components/navigation/Tabs/TabBar.tsx]: Fixed in 54d8b4f7736a62f6952ea8489e07e45cadad1644

File: frontend/components/navigation/Tabs/TabBar.tsx (hunk 5 above)
Comment (Julian702) [reply frontend/components/navigation/Tabs/TabBar.tsx]: Fixed in 54d8b4f7736a62f6952ea8489e07e45cadad1644

File: frontend/components/statistics/Statistic.tsx (hunk 6 above)
Comment (Julian702) [reply frontend/components/statistics/Statistic.tsx]: Fixed in 54d8b4f7736a62f6952ea8489e07e45cadad1644

File: frontend/components/statistics/Statistic.tsx (hunk 7 above)
Comment (Julian702) [reply frontend/components/statistics/Statistic.tsx]: Fixed in 54d8b4f7736a62f6952ea8489e07e45cadad1644

**Review by JJthere — APPROVED** [review]

**Pull request merged**